applications/mne_analyze/plugins/view3d/view3d.h

```diff
@@ -186,19 +192,36 @@ class VIEW3DSHARED_EXPORT View3D : public ANSHAREDLIB::AbstractPlugin
      */
     void settingsChanged(ANSHAREDLIB::View3DParameters viewParameters);
 
-    QPointer<ANSHAREDLIB::Communicator>             m_pCommu;               /**< To broadcst signals */
+    //=========================================================================================================
+    /**
+     * @brief newDipoleFit
+     * @param ecdSet
+     */
+    void newDipoleFit(const INVERSELIB::ECDSet& ecdSet);
+
+    //=========================================================================================================
+    /**
+     * Loads new  model when current loaded model is changed
+     *
+     * @param [in,out] pNewModel    pointer to currently loaded FiffRawView Model
+     */
+    void onModelChanged(QSharedPointer<ANSHAREDLIB::AbstractModel> pNewModel);
+
+    QPointer<ANSHAREDLIB::Communicator>                     m_pCommu;               /**< To broadcst signals */
+
+    int                                                     m_iFiducial;            /**< Currently selected fiducial */
 
-    int                                             m_iFiducial;            /**< Currently selected fiducial */
+    QSharedPointer<DISP3DLIB::Data3DTreeModel>              m_p3DModel;             /**< The 3D model data */
+    DISP3DLIB::BemTreeItem*                                 m_pBemTreeCoreg;        /**< TThe BEM head model of the coregistration plugin. */
+    DISP3DLIB::DigitizerSetTreeItem*                        m_pDigitizerCoreg;      /**< The 3D item pointing to the tracked digitizers. */
+    DISP3DLIB::DigitizerSetTreeItem*                        m_pMriFidCoreg;         /**< The 3D item pointing to the mri fiducials. */
 
-    QSharedPointer<DISP3DLIB::Data3DTreeModel>      m_p3DModel;             /**< The 3D model data */
-    DISP3DLIB::BemTreeItem*                         m_pBemTreeCoreg;        /**< TThe BEM head model of the coregistration plugin. */
-    DISP3DLIB::DigitizerSetTreeItem*                m_pDigitizerCoreg;      /**< The 3D item pointing to the tracked digitizers. */
-    DISP3DLIB::DigitizerSetTreeItem*                m_pMriFidCoreg;         /**< The 3D item pointing to the mri fiducials. */
+    DISP3DLIB::View3D*                                      m_pView3D;              /**< The Disp3D view. */
+    DISPLIB::Control3DView*                                 m_pControl3DView;       /**< The 3D Control view */
 
-    DISP3DLIB::View3D*                              m_pView3D;              /**< The Disp3D view. */
-    DISPLIB::Control3DView*                         m_pControl3DView;       /**< The 3D Control view */
+    bool                                                    m_bPickingActivated;    /**< If Picking is activated*/
 
-    bool                                            m_bPickingActivated;    /**< If Picking is activated*/
+    QList<QSharedPointer<ANSHAREDLIB::DipoleFitModel>>      m_DipoleList;
```

Comment:
Do we need to hold the dipole fit data as a member here as well?

Reply:
I keep track of them here so that we don't add a copy every time we select the model in the data viewer